### PR TITLE
rebuild header with translations

### PR DESCRIPTION
**Pull request by jesperlauridsen**
Now, a few notes to this PR:

I thought about making some small menu-element components, but decided against it for now because of the need for eventlisteners to click on certain elements in the menu. Bouncing events around seems unnessecary when I could achive what I wanted in the same component, without too much hassle or the component being too big.

Also, the menu rebuilds when changing - I could've made a separate function to take care of changing just translations, but that would just add more code to the file - with a minimal win (at least when I was looking it at it today)

Any feedback on this is greatly appreciated. It is, as we discussed, mainly a try to figure out how to proceed with translations in web components. So any input is nice here!

### CODE REVIEW

File: src/components/nav/menu-component.ts [hunk 1]
```diff
@@ -1,16 +1,73 @@
...
+	da: {
+		primary: TranslationItem[];
+		secondary: TranslationItem[];
+	};
+	en: {
+		primary: TranslationItem[];
+		secondary: TranslationItem[];
+	};
+};
+
```
Comment (jorntx) [inline src/components/nav/menu-component.ts:18]: If it is all the same to you I would like to move this down (just before the HTML). I like it when the class declaration and constructor is right up in ya' face when looking through components.

File: src/components/nav/menu-component.ts [hunk 2]
```diff
@@ -19,16 +76,58 @@ class MenuComponent extends HTMLElement {
...
+		if (main) {
+			main.innerHTML = '';
+			this.createSingleMenu(main, this.translation[this.lang === 'da' ? 'da' : 'en'].primary, 'rdl-primary-nav');
+			this.createSingleMenu(main, this.translation[this.lang === 'da' ? 'da' : 'en'].secondary, 'rdl-secondary-nav');
+			const localeSwitcher: HTMLAnchorElement | null = this.shadow.querySelector('#localeSwitcher');
+			localeSwitcher
+				? localeSwitcher.addEventListener('click', (e) => {
+						this.dispatchLocaleSwitch(e);
+				  })
+				: null;
```
Comment (jorntx) [inline src/components/nav/menu-component.ts:100]: Somehow this recreation of the menu doesn't stack in terms of double elements or events, so the web component/browser must somehow clean up on re-render.

File: src/components/nav/menu-component.ts [hunk 3]
```diff
@@ -19,16 +76,58 @@ class MenuComponent extends HTMLElement {
...
 		window.dispatchEvent(new Event('locale-switch'));
 		e.preventDefault();
 	}
+
+	createFullHeaderMenu() {
+		const main: HTMLElement | null = this.shadow.querySelector('.rdl-main-navigation');
+		if (main) {
+			main.innerHTML = '';
+			this.createSingleMenu(main, this.translation[this.lang === 'da' ? 'da' : 'en'].primary, 'rdl-primary-nav');
+			this.createSingleMenu(main, this.translation[this.lang === 'da' ? 'da' : 'en'].secondary, 'rdl-secondary-nav');
```
Comment (jorntx) [inline src/components/nav/menu-component.ts:94]: I will allow it this time but that call to createSingleMenu with the snug ternary providing the locale is balancing on a sharp edge in terms of readability:)

**Review by jorntx — COMMENTED** [review]
I think this is a very good start. The built in types may balloon as we go and force us to externalize them but still keep them bundled with the web component.

I have only one suggestion for improvement in the comments.

File: src/components/nav/menu-component.ts (hunk 2 above)
Comment (jesperlauridsen) [reply src/components/nav/menu-component.ts:100]: main.innerHTML = ''; takes care of this - i remove everything from the div just before i add something new.

File: src/components/nav/menu-component.ts (hunk 1 above)
Comment (jesperlauridsen) [reply src/components/nav/menu-component.ts:18]: Good idea, will do.

**Review by jorntx — APPROVED** [review]
Looks good!

**Pull request merged**